`-Wmaybe-uninitialized` in MethodBind on calls to `ClassDB::bind_vararg_method` with GCC 12 and UBSAN

### Godot version

4.0.beta (f74491fdee9bc2d68668137fbacd8f3a7e7e8df7)

### System information

Mageia 9 Linux, x86_64 | GCC 12.2.1 (20220917)

### Issue description

When building the current `master` branch with GCC 12.2.1 (20220917) and `use_ubsan=yes warnings=extra`, I get the following warning:
```
g++ -o modules/gdscript/gdscript.linuxbsd.tools.x86_64.san.o -c -std=gnu++17 -Wctor-dtor-privacy -Wnon-virtual-dtor -Wplacement-new=1 -g3 -DSANITIZERS_ENABLED -fsanitize=undefined,shift,shift-exponent,integer-divide-by-zero,unreachable,vla-bound,null,return,signed-integer-overflow,bounds,float-divide-by-zero,float-cast-overflow,nonnull-attribute,returns-nonnull-attribute,bool,enum,vptr,pointer-overflow,builtin -fsanitize=bounds-strict -pipe -Wall -Wextra -Wwrite-strings -Wno-unused-parameter -Wshadow-local -Wno-misleading-indentation -Walloc-zero -Wduplicated-branches -Wduplicated-cond -Wstringop-overflow=4 -Wlogical-op -Wattribute-alias=2 -DDEBUG_ENABLED -DDEV_ENABLED -DNO_EDITOR_SPLASH -DTOUCH_ENABLED -DFONTCONFIG_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DDBUS_ENABLED -DSPEECHD_ENABLED -DJOYDEV_ENABLED -DUDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -D_FILE_OFFSET_BITS=64 -DVULKAN_ENABLED -DGLES3_ENABLED -DTOOLS_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -DUSE_VOLK -DVK_USE_PLATFORM_XLIB_KHR -DGLAD_ENABLED -DGLES_OVER_GL -Ithirdparty/libpng -Ithirdparty/glad -Ithirdparty/volk -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/linuxbsd -I. -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libxml2 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/speech-dispatcher modules/gdscript/gdscript.cpp
In file included from ./core/object/class_db.h:34,
                 from ./core/object/ref_counted.h:34,
                 from ./core/io/file_access.h:36,
                 from ./core/io/xml_parser.h:34,
                 from ./core/doc_data.h:34,
                 from ./core/object/script_language.h:34,
                 from ./core/debugger/script_debugger.h:34,
                 from modules/gdscript/gdscript.h:35,
                 from modules/gdscript/gdscript.cpp:31:
./core/object/method_bind.h: In member function 'Variant MethodBindVarArgTR<T, R>::call(Object*, const Variant**, int, Callable::CallError&) [with T = GDScript; R = Variant]':
./core/object/method_bind.h:245:114: warning: '<anonymous>' may be used uninitialized [-Wmaybe-uninitialized]
  245 |                 return (static_cast<T *>(p_object)->*MethodBindVarArgBase<MethodBindVarArgTR<T, R>, T, R, true>::method)(p_args, p_arg_count, r_error);
      |                                                                                                                  ^~~~~~
./core/object/method_bind.h:245:114: note: '<anonymous>' was declared here
  245 |                 return (static_cast<T *>(p_object)->*MethodBindVarArgBase<MethodBindVarArgTR<T, R>, T, R, true>::method)(p_args, p_arg_count, r_error);
      |                                                                                                                  ^~~~~~
```

This is raised by this call:
```
modules/gdscript/gdscript.cpp
1039:   ClassDB::bind_vararg_method(METHOD_FLAGS_DEFAULT, "new", &GDScript::_new, MethodInfo("new"));
```

But it's not specific to GDScript and happens on all `ClassDB::bind_vararg_method` uses.

I don't know yet if there's an actual bug or if it's a false positive from GCC. I'm not expert enough in C++ to know if a function pointer like `&GDScript::_new` could indeed be considered uninitialized (or if it's a natural consequence of UBSAN and not something we should aim to fix).

I tested and reproduced the warning successfully with both `-O0` and `-Og` optimization levels.

CC @godotengine/core @hpvb @marxin 

### Steps to reproduce

- Install GCC 12.2.1 (other versions might also trigger this, Clang untested).
- `p=linuxbsd use_ubsan=yes verbose=yes warnings=extra werror=no -j3 module_raycast_enabled=no module_denoise_enabled=no`
  * Disabling denoise and raycast is not mandatory, it's just that embree and oidn both take *ages* to compile with ubsan.

### Minimal reproduction project

_No response_

--- a/core/object/method_bind.h
+++ b/core/object/method_bind.h
@@ -241,9 +241,17 @@ class MethodBindVarArgTR : public MethodBindVarArgBase<MethodBindVarArgTR<T, R>,
 	friend class MethodBindVarArgBase<MethodBindVarArgTR<T, R>, T, R, true>;
 
 public:
+#if defined(SANITIZERS_ENABLED) && defined(__GNUC__) && !defined(__clang__)
+	// Workaround GH-66343 raised only with UBSAN, seems to be a false positive.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
 	virtual Variant call(Object *p_object, const Variant **p_args, int p_arg_count, Callable::CallError &r_error) override {
 		return (static_cast<T *>(p_object)->*MethodBindVarArgBase<MethodBindVarArgTR<T, R>, T, R, true>::method)(p_args, p_arg_count, r_error);
 	}
+#if defined(SANITIZERS_ENABLED) && defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
 
 	MethodBindVarArgTR(
 			R (T::*p_method)(const Variant **, int, Callable::CallError &),

